### ISSUE TITLE
sql: crash with EXPLAIN (VEC) and DELETE

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#70438]
The following crashes on master (but not on 21.1.7):
```
CREATE TABLE kv (
  k INT PRIMARY KEY,
  v INT,
  UNIQUE INDEX foo (v),
  INDEX bar (k, v)
);
INSERT INTO kv VALUES (1, 2), (3, 4), (5, 6), (7, 8);
EXPLAIN (VEC) DELETE FROM kv WHERE k=3 OR v=6;
```
I'm tentatively adding a GA-blocker issue to investigate the impact (not sure whether only EXPLAIN (VEC) is susceptible or the crash can occur without it too).

**Comment by yuzefovich** [comment]
I think this repro can only occur with `EXPLAIN (VEC)`, but it is a regression with a simple fix, so I'll keep the GA-blocker label and let the release triage meeting decide whether it's ok to merge this for 21.2.0.

**Comment by yuzefovich** [comment]
Reopening so that the release triage meeting decides whether #70466 and #70467 are justifiable to be backported for 21.2.0.

**Comment by yuzefovich** [comment]
Note that this doesn't affect the stmt bundle collection where we're using the already created `opChains` to populate two `explain vec` files. So only explicitly running `EXPLAIN (VEC)` on some statements is affected.

**Comment by yuzefovich** [comment]
Fixed by #70524.
